### PR TITLE
chore: cherry-pick #11977

### PR DESCRIPTION
**Pull request by NicolasMassart**
This PR cherry-picks #11977

### CODE REVIEW

**Review by MarioAslau — APPROVED** [review]

**Comment by tommasini** [comment]
I'm not identifying what but it seems bigger than the one that it was merged in [main](https://github.com/MetaMask/metamask-mobile/pull/11977/files)

File: app/store/index.ts [hunk 1]
```diff
@@ -54,10 +62,19 @@ const createStoreAndPersistor = async () => {
...
+  trace({
+    name: TraceName.StorageRehydration,
+    op: TraceOperation.StorageRehydration,
+  });
+
   /**
    * Initialize services after persist is completed
    */
-  const onPersistComplete = () => {
+  const onPersistComplete = async () => {
```
Comment (tommasini) [inline app/store/index.ts:75]: This shouldn't be async

Comment (NicolasMassart) [reply app/store/index.ts:75]: I;m converting it to draft as it looks there's some work to do on it.

Comment (MarioAslau) [reply app/store/index.ts:75]: fixed

**Review by EtherWizard33 — APPROVED** [review]
LGTM

**Pull request merged**